addons/medical_engine/XEH_postInit.sqf

```diff
@@ -24,7 +24,11 @@
         // "exitWith", which would be broken with "HandleDamage" otherwise.
         _unit setVariable [
             QEGVAR(medical,HandleDamageEHID),
-            _unit addEventHandler ["HandleDamage", {_this call FUNC(handleDamage)}]
+            _unit addEventHandler ["HandleDamage", {
+                // Not returning anything for dead units prevents double execution of Killed events.
+                private _return = _this call FUNC(handleDamage);
+                if (alive _unit) then {_return};
```

Comment:
`_unit` is `nil` in this scope.

Reply:
Sadly, this is what was making it "work", it our EH always just returned nil; which meant the unit just died from vanilla medical.